src/static/helpers/passwordlessHelper.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {ShopperLogin, TokenResponse} from '../../lib/shopperLogin';
+import {createCodeVerifier, stringToBase64} from './slasHelper';
+
+/**
+ * Function to send passwordless login token
+ * **Note** At the moment, passwordless is only supported on private client
+ * @param slasClient a configured instance of the ShopperLogin SDK client.
+ * @param credentials - the id and password and clientSecret (if applicable) to login with.
+ * @param credentials.clientSecret? - secret associated with client ID
+ * @param parameters - parameters to pass in the API calls.
+ * @param parameters.callbackURI? - URI to send the passwordless login token to. Must be listed in your SLAS configuration. Required when mode is callback
+ * @param parameters.usid? - Unique Shopper Identifier to enable personalization.
+ * @param parameters.userid - User Id for login
+ * @param parameters.locale - The locale of the template. Not needed for the callback mode
+ * @param parameters.mode - Medium of sending login token
+ * @returns Promise of Response
+ */
+export async function authorizePasswordless(
+  slasClient: ShopperLogin<{
+    shortCode: string;
+    organizationId: string;
+    clientId: string;
+    siteId: string;
+  }>,
+  credentials: {
+    clientSecret: string;
```

Comment:
You might want to throw if this information is not provided as we are doing [here](https://github.com/SalesforceCommerceCloud/commerce-sdk-isomorphic/blob/10a7405a6594faed7d63a97bc64acc3769003375/src/static/helpers/slasHelper.ts#L187C3-L191C4). This comment also applies to the `getPasswordLessAccessToken` function implementation below.